features/step_definitions/logging.rb

```diff
@@ -705,3 +705,135 @@
+# it seems some options are not supported to 'oc run' command. Instead of that, I use a job to view logs, and delete it later. Leave the 'oc run' steps here for later
+#oc run kafka-consumer -ti --image=registry.redhat.io/amq7/amq-streams-kafka-25-rhel7:1.5.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic topic-logging-app --max-messages 10 --from-beginning
+#step %Q/I run the :run client command with:/,table(%{
+#    | name    | kafka-consumer |
+#    | image   | #{kafka_image} |
+#    | restart | Never |
+#    | command | true  |
+#    | cmd     | bin/kafka-console-consumer.sh  |
+#    | cmd     | --bootstrap-server my-cluster-kafka-bootstrap:9092  |
+#    | cmd     | --topic #{topic_name}   |
+#    | cmd     | --from-beginning  |
+#    | cmd     | --max-messages #{record_num} |
+#    | cmd     | --timeout-ms=300000  |
+#})
+Given /^I get(?: (\d+))? records from the #{QUOTED} kafka topic$/ do | record_num, topic_name|
+  record_num = record_num ? record_num.to_str : "10"
+  job_name=topic_name
+  kafka_image=stateful_set('my-cluster-kafka').containers_spec(user: user)[0].image
+  retries = 3
+  while true
+    step %Q/I process and create:/,table(%{
+      | f | #{BushSlicer::HOME}/testdata/logging/clusterlogforwarder/kafka/amq/21_job_topic_consumer_from_beginning_template.yaml |
+      | p | KAFKA_IMAGE=#{kafka_image} |
+      | p | KAFKA_TOPIC=#{topic_name}  |
+      | p | MAX_MESSAGES=#{record_num} |
+      | p | JOB_NAME=#{job_name} |
+    })
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | job-name=#{job_name}|
+    })
+    sleep 30
+    step %Q/I run the :logs client command with:/, table(%{
+      | resource_name | #{ pod.name} |
+    })
+
+    admin.cli_exec(:delete, object_type: 'job', object_name_or_id: job_name, n: cb.kafka_project.name)
+    if @result[:response].match("pipeline_metadata")
+      @result[:success] = true
+      break
+    elsif retries == 0
+      @result[:success] = false
+      break
+    end
+    sleep 30
+    retries -= 1
+  end
+end
+
+# Register an topic consumer which listen the topic continuously. we can verify the logging records by the next step 'I get N logs from the X kafka consumer job'
+Given /^I listen on the #{QUOTED} kakafa topic using the #{QUOTED} job$/ do | topic_name, job_name |
+  kafka_image=statefulset('my-cluster-kafka').containers_spec(user: user)[0].image
+  step %Q/I process and create:/,table(%{
+    | f | "#{BushSlicer::HOME}/testdata/logging/clusterlogforwarder/kafka/amq/21_job_topic_consumer_from_latest_template.yaml" |
+    | p | KAFKA_IMAGE=#{kafka_image} |
+    | p | KAFKA_TOPIC=#{topic_name} |
+    | p | MAX_MESSAGES=#{record_num} |
+    | p | JOB_NAME=#{job_name} |
+  })
+end
+
+# Get logs from the kafka comsumer job which created by upper job
+Given /^I get(?: (\d+))? logs from the #{QUOTED} kafka consumer job$/ do | record_num, job_name |
```

Comment:
I have same questions as step `I get(?: (\d+))? records from the #{QUOTED} kafka topic`